### PR TITLE
Args are stored from previous execs but not used for the next launch

### LINKED ISSUE #3694: Next execution of scheduled task adds arguments from previous task execution
**Description:**
If task is scheduled and then launched manually with new argument, next scheduled execution will have the same argument. I discovered this after restarting batch job which creates new task execution with additional incrementable argument. Next time Kubernetes job was created, new task execution had the same argument which resulted in error (job instance already complete with provided parameters).

**Release versions:**
Spring Cloud Data Flow 2.3.0.RELEASE on Kubernetes

**Steps to reproduce:**
1. Create task and schedule it.
2. Launch same task with some arguments.
3. Next time scheduled task execution is created it will have the same arguments.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #3694

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java [hunk 1]
```diff
@@ -254,10 +254,9 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 
 		// We now have a new props and args what should really get used.
 		Map<String, String> mergedTaskDeploymentProperties = report.getMergedDeploymentProperties();
-		List<String> mergedCommandLineArguments = report.getMergedCommandLineArguments();
 
 		taskExecutionInformation.setTaskDeploymentProperties(mergedTaskDeploymentProperties);
-		appDeploymentRequest = updateDeploymentProperties(mergedCommandLineArguments, platformName, taskExecutionInformation,
+		appDeploymentRequest = updateDeploymentProperties(commandLineArgs, platformName, taskExecutionInformation,
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionService.java]: I wonder how and why was the `mergedCommandLineArguments` used before. By any chance, can we get this `executeTask` method refactored as this has lots of moving parts and becoming incomprehensible?

**Comment by cppwfs** [comment]
Hello Ilaya,
I can break this method up into some manageable pieces .
As to how `mergedCommandLineArguments` was used in the past.   In this case it was not used in previous releases of Spring Cloud Data Flow.   It was introduced in the following commit: https://github.com/spring-cloud/spring-cloud-dataflow/commit/1ce362a23679a54ef24241c503de0078f4f92f1c#diff-ecf6ab80ac41cacc755033ef4163b11f.  However after discussions with @mminella it was determined that this was done in error and should not have been added.

**Comment by cppwfs** [comment]
PR Updated.  Broke up the method so that it is more readable.

**Comment by sabbyanandan** [comment]
@cppwfs: Does this PR also address https://github.com/spring-cloud/spring-cloud-dataflow/issues/3695?

**Comment by mminella** [comment]
Squashed, rebased, and merged as aea7d207a77ccf8570c1bfed753bc2e107f051f1

**Comment by canerkoseren** [comment]
Hi all,

I was using SCDF 2.5.1 and having a problem like launching a task. I could launch the application with the properties of previous execution. 

Today I updated the version of scdf to 2.7.1 after seeing this fix & merge, but the result was same. 

By using scdf-dashboard, I executed the first application with the properties given below
```
deployer.batchapp.kubernetes.environmentVariables=PARAMS1=input_file1.csv,PARAMS2=output_file2.csv
deployer.batchapp.kubernetes.readinessProbeDelay=120
deployer.batchapp.kubernetes.livenessProbePath=/live
deployer.batchapp.kubernetes.readinessProbePath=/ready
deployer.batchapp.kubernetes.livenessProbeDelay=120
```

After the application ends without error, I wanted to launch the task again. On the scdf-dashboard, I opened the task page and click "Launch Task", and in the next page, I saw all parameters of the previous execution.

Regarding to this fix, shouldn't that parameter section be opened empty?

**Comment by cppwfs** [comment]
Helllo  @canerkoseren ,
This fix pertains to command line arguments of a task launch, not the properties.   The properties will be carried forward for each new launch of a task.

**Comment by canerkoseren** [comment]
Hi @cppwfs, thanks for replies.

**Pull request closed without merging**